### ISSUE TITLE
Stream iteration blocking?

### ISSUE THREAD

**Issue by gusutabopb** [vxgmichel/aiostream#4]
```python
import asyncio
import aiostream

async def counter():
    s = aiostream.stream.count(interval=0)
    async with s.stream() as stream:
        async for event in s:
            print(event)
            
async def foo():
    while True:
        print('foo')
        await asyncio.sleep(0.01)

loop = asyncio.get_event_loop()
fut = asyncio.wait([counter(), foo()])
loop.run_until_complete(fut)
```

Running the above code, `counter` seems to block the event loop indefinitely and `foo` never gets called:
```bash
$ python aiostream_issue.py > aiostream_issue.log
$ cat aiostreamer_bug.log | grep foo | wc
       0       0       0
```
However, I found a couple of workarounds:
1) Make `interval` positive
2) Pipe `s` with `pipe.spaceout(interval=0)`
3) Add `await asyncio.sleep(0)` after `print(event)` 

Applying any of the above workarounds and running again for a few seconds:
```bash
$ python aiostream_issue.py > aiostream_issue.log
$ cat aiostreamer_bug.log | grep foo | wc
     425     425    1700
```

I have tried replacing `asyncio.wait` with `asyncio.gather`, using `uvloop`, and other minor changes, but the results were the same.

I thought that a `Stream` object would implicitly yield the control flow to the the event loop after each iteration (as is the case in workaround 2), but that seems to not always be the case. Is this by design or a perhaps a bug?

PS: Thanks a lot for this great library!

**Comment by vxgmichel** [comment]
Thanks for the report!

> I thought that a Stream object would implicitly yield the control flow to the the event loop after each iteration (as is the case in workaround 2), but that seems to not always be the case. Is this by design or a perhaps a bug?

Not yielding after each iteration is by design, but the issue you describe is definitely a bug. I guess we could give the control back to the event loop right before yielding the value for all the stream operators (in `Streamer.__anext__` for instance), but I'm afraid it would affect the performance too much. Instead, I'd rather fix `from_iterable` (see commit 514759e).

Do you need a new release for this fix?

**Comment by gusutabopb** [comment]
Thanks for the quick fix and explanation!

Actually, it seems I found this bug by accident as the bug in my code was caused by this misconception I had about the behavior of `Stream`s (and `AsyncIterable`s in general). I am not using `counter` or `from_iterable` in my program. I am making a `Stream` object using `merge` by passing a regular `AsyncIterable`s and I was missing an `await sleep(0)` in a very specific location inside one of those `AsyncIterable`s.

As for the new release, it won't really make any difference to me it so will leave it up to you.

Feel free to close this issue :)
